### PR TITLE
[FIX] survey: conditional score with sections

### PR DESCRIPTION
**Pull request by backspac**
Steps:
- Go to Surveys
- Create an new survey:
  - Questions:
    1. First section
    2. First question:
       - Multiple choice: only one answer
       - Answers:
         1. First answer
            - Choice: Yes
            - Is a correct answer: Checked
            - Score for this choice: 1
         2. Second answer
            - Choice: No
    3. Second question:
       - Multiple choice: only one answer
       - Answers:
         1. First answer
            - Choice: Yes
            - Is a correct answer: Checked
            - Score for this choice: 1
         2. Second answer
            - Choice: No
       - Options tab:
         - Conditional display: Checked
         - Triggering question: (First question)
         - Triggering answer: (First question, First answer)
  - Options tab:
    - Layout: One page per section
    - Scoring: Scoring with answers at the end
- Click Test
- Select the two correct answers
- Submit

Bug:
The score is 50% instead of 100%.

Explanation:
Since multiple questions are submitted at the same time, we cannot know
before saving the question whether the following is still going to be
active.

This commit re-evaluates, for each question, which are the ones that are
still active.

opw:2537713

### CODE REVIEW

**Comment by backspac** [comment]
@dbeguin Can I ask for your input on this one?

File: addons/survey/controllers/main.py [hunk 1]
```diff
@@ -467,10 +467,7 @@ def survey_submit(self, survey_token, answer_token, **post):
 
         errors = {}
         # Prepare answers / comment by question, validate and save answers
-        inactive_questions = request.env['survey.question'] if answer_sudo.is_session_answer else answer_sudo._get_inactive_conditional_questions()
         for question in questions:
-            if question in inactive_questions:  # if question is inactive, skip validation and save
-                continue
```
Comment (dbeguin) [inline addons/survey/controllers/main.py:473]: I think this should be done like that instead : 
```
for question in questions:            
    inactive_questions = request.env['survey.question'] if answer_sudo.is_session_answer else answer_sudo._get_inactive_conditional_questions()    
    if question in inactive_questions:  # if question is inactive, skip validation and save                
        continue
```
The idea is to re-evaluate which question is active or not, after submitting question per question. In page per question, this is not mandatory as only one question is submitted one by one. For the two other cases, as multiple questions are submitted at the same time, we cannot know before saving the question if the following will still be active or not.

removing the check completely is dangerous as we assume that the frontend is doing all the job of filtering inactive questions before submitting - which is the case - but by security, we should align both behaviour at frontend and backend side.

**Comment by Yenthe666** [comment]
The question that brings me up here is: could this have impacted people trying for the official Odoo exam? Where they could have passed it but didn't because of this? 🤔

**Comment by dbeguin** [comment]
> The question that brings me up here is: could this have impacted people trying for the official Odoo exam? Where they could have passed it but didn't because of this?

@Yenthe666 : No, because there is no conditional question in Odoo Certifications.

**Review by dbeguin — APPROVED** [review]
LGTM

**Comment by backspac** [comment]
@nle-odoo I think this one can be merged :)

**Review by nle-odoo — APPROVED** [review]
robodoo r+

**Pull request closed without merging**